sdk/search/search-documents/src/serviceUtils.ts

```diff
@@ -521,6 +524,57 @@ export function publicIndexToGeneratedIndex(index: SearchIndex): GeneratedSearch
     similarity: convertSimilarityToGenerated(index.similarity)
   };
 }
+export function generatedIndexerStatusToPublicIndexerStatus(
+  generatedSearchIndexerStatus: GeneratedSearchIndexerStatus
+): SearchIndexerStatus {
+  let lastResult;
+  if (generatedSearchIndexerStatus.lastResult) {
+    lastResult = {
+      ...generatedSearchIndexerStatus.lastResult,
+      currentState: generatedSearchIndexerStatus.lastResult.currentState
+        ? {
+            ...generatedSearchIndexerStatus.lastResult.currentState,
+            allDocumentsInitialChangeTrackingState:
```

Comment:
couldn't we do this with x-ms-clientname renames instead of having to write runtime code to deserialize it differently?

Reply:
Done